metron/core/llm_clients/openai_chat_completions_client.py

```diff
@@ -82,56 +80,61 @@ def send_llm_request(
         most_recent_received_token_time = time.monotonic()
 
         try:
-            with requests.post(
-                address,
-                json=body,
-                stream=True,
-                timeout=180,
-                headers=headers,
-            ) as response:
-                if response.status_code != 200:
-                    error_msg = response.text
-                    error_response_code = response.status_code
-                    logger.error(f"Request Error: {response.content}")
-                    response.raise_for_status()
-
-                for chunk in response.iter_lines(chunk_size=None):
-                    chunk = chunk.strip()
-
-                    if not chunk:
-                        continue
-                    stem = "data: "
-                    chunk = chunk[len(stem) :]
-                    if chunk == b"[DONE]":
-                        continue
-                    data = json.loads(chunk)
-
-                    if "error" in data:
-                        error_msg = data["error"]["message"]
-                        error_response_code = data["error"]["code"]
-                        raise RuntimeError(data["error"]["message"])
-
-                    delta = data["choices"][0]["delta"]
-                    if delta.get("content", None):
-                        (
-                            current_tokens_received,
-                            previous_token_count,
-                        ) = self.get_current_tokens_received(
-                            previous_responses=previous_responses,
-                            current_response=delta["content"],
-                            previous_token_count=previous_token_count,
-                        )
-
-                        tokens_received += current_tokens_received
-                        inter_token_times.append(
-                            time.monotonic() - most_recent_received_token_time
-                        )
-                        if current_tokens_received > 1:
-                            inter_token_times.extend(
-                                [0] * (current_tokens_received - 1)
+            async with httpx.AsyncClient() as client:
```

Comment:
nit: do we need to create a new client everytime?

Reply:
Yes, we don't need create client every time. Instead we can initialize client once inside constructor and then using it repeatedly during each send_llm_request() call.